src/Sarif.Sarifer/GenerateTestDataCommand.cs

```diff
@@ -9,53 +9,61 @@
 using Microsoft.Sarif.Viewer.Interop;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.CodeAnalysis.Sarif.Sarifer
 {
-    internal class GenerateTestDataCommand : MenuCommand
+    internal class GenerateTestDataCommand : SariferCommandBase
     {
-        private static SarifViewerInterop s_viewerInterop;
+        private const string SendDataToViewerFailureEventName = "SendDataToViewer/Failure";
 
-        public GenerateTestDataCommand(IVsShell vsShell) :
-            base(
-                new EventHandler(MenuCommandCallback),
-                new CommandID(Guids.SariferCommandSet, SariferPackageCommandIds.GenerateTestData))
+        private readonly SarifViewerInterop viewerInterop;
+
+        public GenerateTestDataCommand(IVsShell vsShell, IMenuCommandService menuCommandService)
         {
-            s_viewerInterop = new SarifViewerInterop(vsShell);
+            this.viewerInterop = new SarifViewerInterop(vsShell);
+
+            MenuCommand menuCommand = new MenuCommand(
+                new EventHandler(this.MenuCommandCallback),
+                new CommandID(Guids.SariferCommandSet, SariferPackageCommandIds.GenerateTestData));
+
+            menuCommandService.AddCommand(menuCommand);
         }
 
         /// <summary>
         /// Event handler called when the user selects the Generate SARIF Test Data command.
         /// </summary>
-        /// <remarks>
-        /// Since this is a menu item callback, it must return void.
-        /// </remarks>
-#pragma warning disable VSTHRD100 // Avoid async void methods
-        private static async void MenuCommandCallback(object caller, EventArgs args)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+        private void MenuCommandCallback(object caller, EventArgs args)
         {
-            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            this.SendDataToViewerAsync().FileAndForget(GetFileAndForgetEventName(SendDataToViewerFailureEventName));
+        }
 
-            string testDataFilePath = await CreateTestDataFileAsync();
+        private async Task SendDataToViewerAsync()
+        {
+            string testDataFilePath = await CreateTestDataFileAsync().ConfigureAwait(continueOnCapturedContext: true);
```

Comment:
>true [](start = 112, length = 4)

This is interesting. One would thing that the SARIF viewer code would switch to the UI thread if it needed to :)

Can't think of a reason why either of these need to be true.. (There may be one, I just don't know it)

Reply:
I'm convinced. Changed.

---
In reply to: [514645936](https://github.com/microsoft/sarif-visualstudio-extension/pull/238#discussion_r514645936) [](ancestors = 514645936)